nextflow.config

```diff
@@ -1,40 +1,149 @@
+ * name here.
+ */
+
+// Global default params, used in configs
+params {
+    // Workflow flags
+    genome = false
+    reads = "data/*{1,2}.fastq.gz"
+    singleEnd = false
+    outdir = './results'
+    supernova_container = 'nfcore/supernova:latest'
+    clusterOptions = false
+    minsize = 1000
+    full_output = false
+    busco_folder = "$baseDir/data"
+    test = false
+    reads = false
+    lanes = false
+    indices = false
+    maxreads = false
+    bcfrac = false
+    nopreflight = false
+    no_accept_extreme_coverage = false
+    fasta = false
+    genomesize = false
+    genomes = false
+    project = false
+
+    // Boilerplate options
+    name = false
+    multiqc_config = "$baseDir/assets/multiqc_config.yaml"
+    email = false
+    email_on_fail = false
+    maxMultiqcEmailFileSize = 25.MB
+    plaintext_email = false
+    monochrome_logs = false
+    help = false
+    igenomes_base = "./iGenomes"
+    tracedir = "${params.outdir}/pipeline_info"
+    awsqueue = false
+    awsregion = 'eu-west-1'
+    igenomesIgnore = true
+    custom_config_version = 'master'
+    custom_config_base = "https://raw.githubusercontent.com/nf-core/configs/${params.custom_config_version}"
+    hostnames = false
+    config_profile_description = false
+    config_profile_contact = false
+    config_profile_url = false
+}
+
+timeline {
+  enabled = true
+  file = "${params.tracedir}/execution_timeline.html"
+}
+report {
+  enabled = true
+  file = "${params.tracedir}/execution_report.html"
+}
+trace {
+  enabled = true
+  file = "${params.tracedir}/execution_trace.txt"
+}
+dag {
+  enabled = true
+  file = "${params.tracedir}/pipeline_dag.svg"
+}
+
+process.container = 'nfcore/neutronstar:1.0.0'
+
+// Load base.config by default for all pipelines
+includeConfig 'conf/base.config'
 
-    standard {
-        process.executor = 'local'
-        process.cpus = { 8 }
-        process.memory = { 64.GB }
-        process.time = { 1.h * task.attempt }
-        process.errorStrategy = { task.exitStatus == 143 ? 'retry' : 'finish' }
-        process.maxRetries = 2
-        process.maxErrors = '-1'
+profiles {
+    awsbatch { includeConfig 'conf/awsbatch.config' }
+    conda { process.conda = "$baseDir/environment.yml" }
+    debug { process.beforeScript = 'echo $HOSTNAME' }
+    docker {
+        docker.enabled = true
+        fixOwnership = true
+        runOptions = "-u \$(id -u):\$(id -g)"
     }
+    singularity { singularity.enabled = true }
+    test { includeConfig 'conf/test.config' }
+    slurm { includeConfig 'conf/slurm.config' }
+
+}
+
+// Avoid this error:
+// WARNING: Your kernel does not support swap limit capabilities or the cgroup is not mounted. Memory limited without swap.
+// Testing this in nf-core after discussion here https://github.com/nf-core/tools/pull/351, once this is established and works well, nextflow might implement this behavior as new default.
+docker.runOptions = '-u \$(id -u):\$(id -g)'
```

Comment:
```suggestion

```

Reply:
Why is this not needed? I'm only asking since I really don't know what it does 😁

Reply:
A bit further up you already have it inside the scope - no need to duplicate this here:

```
  docker {
        process.maxErrors = '-1'
        docker.enabled = true
        fixOwnership = true
        runOptions = "-u \$(id -u):\$(id -g)"
    }
    }
```

Reply:
Oh of course 🤦‍♂